bin/annotate_test_failures

```diff
@@ -100,6 +115,76 @@ def update_annotation(title, list, style, state)
   end
 end
 
+# Given a list of failures, parse list and send a slack notification with the test names in the payload
+#
+def send_slack_notification(slack_webhook, slack_channel, title, list)
+  tests = list.map do |item|
+    {
+      'name' => item.name,
+      'classname' => item.classname,
+      'message' => item.message,
+      'details' => item.details
+    }
+  end
+
+  if !["trunk", "develop", "master"].include?(ENV['BUILDKITE_BRANCH']) # Don't send notification if build is not on trunk/develop/master branch
+    puts "⏭  Failures not on trunk/develop/master branch. Slack notification not sent."
+  else
```

Comment:
This definitively shouldn't be hardcoded here. This tool is a general-purpose utility that might be called by any BuildKite project that generates Junit results. We can't assume which repos will be using it and which branches the users of that script will want to enable or disable Slack notifications for.

For example, maybe at some point some of our pods repos like WPAuthenticator and friends, or even a brand new repo, will like to use this utility. There's no guarantee that those other repos will use those branch names… nor that we'd want the Slack notification not to be sent for them on trunk (in fact, I can totally see some repos willing to send the notification on test failures even for builds on trunk)

Therefore that decision to not send the Slack notification on certain conditions (like branch betting trunk or not) should be made by the callers of this script, not hardcoded in the scripts itself. For example WCiOS could decide to use:
```bash
if [[ "$BUILDKITE_BRANCH" =~ ^(develop|trunk)$ ]]; then
  annotate_test_failures
else
  annotate_test_failures --slack 'build-and-ship'
fi
```